beer-garden/functionary
# #114 - Scheduled Tasking frontend support

Closes #114 

## Introduced Changes
- Added scheduled task list, create, and update pages
- Dynamically load function parameters with htmx
- Dynamically validate crontab fields with htmx
- Added separate htmx urls
- Display message to user if they have no scheduled tasks they can make in their current environment
- Display form errors when creating or updating a `ScheduledTask`


## Testing
- Launch Functionary and view the `Scheduling` UI
- Create tasks and verify they run according to the crontab schedule
- Update tasks to verify their next run reflects those updates
- Try and submit invalid data to the forms and verify you receive an error


## Notes
- There are some helper functions in `core.utils.scheduling` that I think can be migrated over to the `ScheduledTask` model
- With my current implementation, I do not persist parameter arguments when a form gets rejected. The parameter values will reset to their defaults according to their widget type
